Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.Streetcode.TextContent;
+using Streetcode.BLL.MediatR.Streetcode.Text.GetAll;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Streetcode.XUnitTest.StreetcodeTest.TextTest
+{
+    public class GetAllTextsTest
+    {
+        private Mock<IRepositoryWrapper> repository;
+        private Mock<IMapper> mockMapper;
+
+        public GetAllTextsTest()
+        {
+            repository = new Mock<IRepositoryWrapper>();
+            mockMapper = new Mock<IMapper>();
+        }
+
+        [Fact]
+        public async Task ShouldReturnSuccesfullyAllTexts()
+        {
+
+            var testTextsList = new List<Text>()
+        {
+            new Text() { Id = 1 }
+                   };
+
+            var testTextslistDTO = new List<TextDTO>()
+        {
+            new TextDTO() { Id = 1 }
+        };
+
+            var testTexts = new Text() { Id = 1 };
+
+            repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(), It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+                .ReturnsAsync(testTexts);
+
+            repository.Setup(repo => repo.TextRepository.GetAllAsync(null, null)).ReturnsAsync(testTextsList);
+
```

Comment:
why two repository setups?? the last hides the first one. Also move them to private methods)